idb/postgres/postgres.go

```diff
@@ -85,6 +86,30 @@ func openPostgres(db *sql.DB, opts *idb.IndexerDbOptions, logger *log.Logger) (p
 	return
 }
 
+func txnWithRetry(f func() error) error {
+	maxDelay := time.Minute
+	delay := time.Millisecond
+
+	for {
+		err := f()
+
+		// If not serialization error.
+		var pqerr *pq.Error
```

Comment:
Is there any alternative to exposing pq here? As discussed we probably should replace this library and this makes it a little more difficult.

Reply:
I think the only alternative is too compare the error message. Not pretty.